Error 500 internal server error when trying to print data table

jonrandahl opened the pull request:
This PR resolves the error reported in ticket #396 by updating the print presenter to use [`.try(:first)`](https://apidock.com/rails/Object/try) which resolves by  returning `nil` without failing if the requested element does not have the  method `.first`, i.e. is empty or nil

## Additional Changes:

- Updated the print template to include the Google Analytics tracking script for the print page as well as importing shared header content for unification of presentation
- Minor tweaks to the Makefile to remove duplicate variables (`SHORTNAME`) as well as introducing new targets for `check` and `local` for streamlined development tasks, alongside removing incorrect and erroneous commands where applicable
- Reorganised .gitignore file to improve readability as well as added comments to group and explain rule application
- Removed duplicated rubocop command
- Reorganised gemfile to create test specific group and lock gem versions to current version of Ruby in use
- Minor Rubocop fixes including preventing rubocop from parsing the `bin/s-query` file which is  used to resolve the fallback location for the Apache Jena Fuseki s-query file added in this [commit](https://github.com/epimorphics/ukhpi/commit/6ba7c50d2d5e7a039f2c462bfd9be659be016b4b)

jonrandahl committed:
fix: resolves failing first calls

using `.try(:first)` resolves by returning `nil` without failing if the requested element does not have the method `.first`, i.e. is empty

jonrandahl committed:
refactor; include tracking in print jobs

Adjusted template to include google analytics in print requests as well as carried over other improvements in header elements

jonrandahl committed:
build: improve make targets

Minor tweaks to the Makefile to remove duplicate variables (`SHORTNAME`) as well as introducing new targets for `check` and `local` for streamlined development tasks

jonrandahl committed:
build: updated patch version cadence

from v1.7.3 ~> v1.7.4

jonrandahl committed:
docs: Updated CHANGELOG

jonrandahl committed:
chore: fix mis-type

removed `:unit` from test target as that doesn't exist

bogdanadrianmarc reviewed (approved):
LGTM!

jonrandahl committed:
style: rubocop fixes

jonrandahl committed:
build: updated and organised .gitignore

jonrandahl committed:
build: gemfile and gemfile.lock updates

Organised specific gems into test group as well as locked gem versions to the current ruby version where applicable

jonrandahl committed:
build: Makefile cleanup

Removed incorrect instructions, re-grouped build commands into the assets target, applied the same target where previously invoking manually; finally, updated the test target to use rake test and removed the system test command as it's not required at this moment

bogdanadrianmarc commented on bin/s-query:
```diff
@@ -235,9 +237,9 @@ def response_no_body(uri, request)
       end
     end
-  rescue Exception => e  
-    # puts e.message  
-    #puts e.backtrace.inspect  
+  rescue Exception => e
+    # puts e.message
+    #puts e.backtrace.inspect
```
Should the comments be removed here? Otherwise looks good to me!

jonrandahl replied:
Good point however those comments are straight from the [copied export from Jena](https://github.com/epimorphics/ukhpi/commit/6ba7c50d2d5e7a039f2c462bfd9be659be016b4b), not manually changed so I'd like to leave them until otherwise required!

jonrandahl committed:
fix: production config to not use ENV.fetch()

jonrandahl committed:
build: removed `local` target

jonrandahl committed:
build: updated production config

set `config.cache_classes=false` to allow Spring to reload assets
set `ENV['API_SERVICE_URL']` to `ENV.fetch('API_SERVICE_URL', nil)` as Rubocop continued to complain

jonrandahl committed:
build: Makefile updates

removes auth target, as well as echos for tracking, alongside reassigning NAME instead of REPO where applicable

jonrandahl commented:
Closing as releasing change through the stack to verify and ensure it's updated on all instances going forward.
